soroban-env-host/src/auth.rs

```diff
@@ -2084,15 +2084,10 @@ impl Host {
         Ok(self
             .try_borrow_previous_authorization_manager_mut()?
             .as_mut()
-            .ok_or_else(|| {
-                self.err(
-                    ScErrorType::Auth,
-                    ScErrorCode::InvalidAction,
-                    "previous invocation is missing - no auth data to get",
-                    &[],
-                )
-            })?
-            .get_authenticated_authorizations(self))
+            .map(|am| am.get_authenticated_authorizations(self))
```

Comment:
This is a bit surprising - the intent here was that we only return error when there was no previous call (i.e. if you call this without calling any contracts at all). After the first invocation previous auth manager has to always be present. I don't mind approving this, but we might have a bit deeper issue here if the auth manager has really been missing after some invocations.

Reply:
Ah so there folks can write SDK tests that never call a contract, and the Env Drop has no way of really knowing if that occurred or not, so it calls .auth() regardless.

Imo it should be fine to call .auth() without a contract call happening, and for it to return an empty list.

Reply:
> Ah so there folks can write SDK tests that never call a contract
Ok, that's fine then. I was just worried that we might trigger this in between contract calls.

Reply:
Would it be bad if we calls .auth() between contract calls?